tests/components/history_stats/test_sensor.py

```diff
@@ -1465,6 +1465,125 @@ def _fake_states(*args, **kwargs):
+                ),
+            ]
+        }
+
+    # The test begins at 23:00, and queries from the database that the sensor has been on since 12:00.
+    with (
+        patch(
+            "homeassistant.components.recorder.history.state_changes_during_period",
+            _fake_states,
+        ),
+        freeze_time(start_time),
+    ):
+        await async_setup_component(
+            hass,
+            "sensor",
+            {
+                "sensor": [
+                    {
+                        "platform": "history_stats",
+                        "entity_id": "binary_sensor.state",
+                        "name": "sensor1",
+                        "state": "on",
+                        "start": "{{ today_at() }}",
+                        "end": "{{ now() }}",
+                        "type": "time",
+                    }
+                ]
+            },
+        )
+        await hass.async_block_till_done()
+
+        await async_update_entity(hass, "sensor.sensor1")
+        await hass.async_block_till_done()
+
+    assert hass.states.get("sensor.sensor1").state == "11.0"
+
+    # Advance 30 minutes
+    t1 = start_time + timedelta(minutes=30)
+    with freeze_time(t1):
+        async_fire_time_changed(hass, t1)
+        await hass.async_block_till_done()
+
+    assert hass.states.get("sensor.sensor1").state == "11.5"
+
+    # Advance 29 minutes, to record the last minute update just before midnight, just like a real system would do.
+    t2 = t1 + timedelta(minutes=29, microseconds=300)
+    with freeze_time(t2):
+        async_fire_time_changed(hass, t2)
+        await hass.async_block_till_done()
+
+    assert hass.states.get("sensor.sensor1").state == "11.98"
+
+    # One minute has passed and the time has now rolled over into a new day, resetting the recorder window. The sensor will then query the database for updates,
+    # and will see that the sensor is ON starting from midnight.
+    t3 = t2 + timedelta(minutes=1)
+
+    def _fake_states_t3(*args, **kwargs):
+        return {
+            "binary_sensor.state": [
+                ha.State(
+                    "binary_sensor.state",
+                    "on",
+                    last_changed=t3.replace(
+                        hours=0, minutes=0, seconds=0, microseconds=0
+                    ),
+                    last_updated=t3.replace(
+                        hours=0, minutes=0, seconds=0, microseconds=0
+                    ),
+                ),
+            ]
+        }
+
+    with (
+        patch(
+            "homeassistant.components.recorder.history.state_changes_during_period",
+            _fake_states_t3,
+        ),
+        freeze_time(t3),
+    ):
+        # The sensor turns off around this time, before the sensor does its normal polled update.
+        hass.states.async_set("binary_sensor.state", "off")
+        await hass.async_block_till_done(wait_background_tasks=True)
```

Comment:
I am struggling with this point in the test.

I update the tracked binary sensor, which causes an update to the history stats sensor, and the history stats sensor executes `_async_history_from_db` as part of that update. 

But this `await` does not wait for that call to return, the test just marches ahead with the history stats update never finishing. 

What can I `await` that waits for that executor job to finish?